ratpack-core/src/test/groovy/ratpack/http/client/HttpForwardProxySpec.groovy

```diff
@@ -86,4 +93,78 @@ class HttpForwardProxySpec extends BaseHttpClientSpec {
     pooled << [true, false]
   }
 
+  def "can make simple proxy request with proxy authentication (pooled: #pooled)"() {
+    given:
+    AtomicBoolean authenticated = new AtomicBoolean(false)
+    AtomicBoolean proxied = new AtomicBoolean(false)
+    def username = "testUser"
+    def password = "testPassword"
+
+    proxyApp {
+      all {
+        def authHeader = request.headers.get(PROXY_AUTHORIZATION)
+        if (authHeader == null) {
+          if (authenticated.get()) {
+            // The client was already authenticated. In real life, we would have validated the client's session.
```

Comment:
When developing this test, I discovered that Netty sends to requests to the HTTP Proxy: the first contains the Proxy-Authorization header, while the second does not.

Rather than adding session handling and such, it was easier in this test to just track when the first request was authenticated and accepted.

Reply:
ACK.